codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/QueryShapeSerVisitor.java

```diff
@@ -160,6 +168,21 @@ private void serializeNamedMember(
         }
     }
 
+    /**
+     * Retrieves the correct serialization location based on the member's
+     * xmlName trait or uses the default value.
+     *
+     * @param memberShape The member being serialized.
+     * @param defaultValue A default value for the location.
+     * @return The location where the member will be serialized.
+     */
+    protected String getMemberSerializedLocationName(MemberShape memberShape, String defaultValue) {
+        // Use the @xmlName trait if present on the member, otherwise use the member name.
+        return memberShape.getTrait(XmlNameTrait.class)
```

Comment:
Should this use ec2QueryName and capitalize the xmlName trait value?

Reply:
This is for standard `aws.query`, the method is overridden in `Ec2QueryShapeSerVisitor` to handle these `aws.ec2` specific customizations.